iree/turbine/kernel/wave/utils.py

```diff
@@ -389,11 +389,11 @@ def add_reshape_if_needed(mma: MMA, prev_mma: MMA, arg_index: int):
             arg = get_custom(arg)
             if is_reshape_needed(arg, mma.vector_shapes, prev_mma.vector_shapes):
                 reshape = Reshape(arg.fx_node, prev_mma.vector_shapes).add_to_graph(
-                    custom.graph
+                    mma.graph
                 )
                 custom_reshape = get_custom(reshape)
-                custom_reshape.vector_shapes = custom.vector_shapes
-                custom.update_arg(arg_index, reshape)
```

Comment:
was this a bug?

Reply:
In this context, yes. It was using custom which is defined in the outside function which was not the same as mma.